tests/e2e/support/objects/app-files/share/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { errors, Page } from 'playwright'
+import util from 'util'
+
+const acceptedShareList =
+  '//*[@data-test-resource-name="%s"]/ancestor::tr//span[@data-test-user-name="%s"]'
+export const acceptedShareExists = async ({
+  page,
+  resource,
+  owner,
+  timeout = 500
+}: {
+  page: Page
+  resource: string
+  owner: string
+  timeout?: number
+}): Promise<boolean> => {
+  let exist = true
+  await page
+    .waitForSelector(util.format(acceptedShareList, resource, owner), { timeout })
+    .catch((e) => {
+      if (!(e instanceof errors.TimeoutError)) {
+        throw e
+      }
+
+      exist = false
+    })
+
+  return exist
+}
```

Comment:
do we need to make explicit file to make this function? may be we can implement in function `isAcceptedSharePresent` in this function

Reply:
And also check if we can reuse this function `resourceExists()` instead of this `acceptedShareExists()` this function.

Reply:
No, we can't reuse resource exists () because it just checks resource existence but we need to check resource existence in accepted share. Yes,i think we need to make an explicit file for a such helper function. I can see a similar pattern used in overall e2e.